app/controllers/application_controller.rb

```diff
@@ -58,7 +58,7 @@ def decorated_user
   end
 
   def after_sign_in_path_for(user)
-    stored_location_for(user) || session[:saml_request_url] || profile_path
+    stored_location_for(user) || profile_path
```

Comment:
Gotta admit, I was able to fix this (specs and local QA confirm that this works) but I am not sure exactly what is going on.

`stored_location_for` comes from devise. It is looking at `session[:user_return_to]`

As far as I can tell, devise does not store this for us anywhere. We do call it once, in `app/controllers/reauthn_required_controller.rb`, but that is not invoked by the saml auth controller, as far as I can tell. Anyone have ideas on how we can improve / clarify this?